AGENTS.md

```diff
@@ -0,0 +1,37 @@
+# BuildBuddy Coding Agent Guidelines
```

Comment:
Most of this guide seems usable as general contribution guidelines (for humans as well). Maybe we'd want to title this "BuildBuddy contribution guidelines" and put it in `CONTRIBUTING.md`.

In terms of these various AGENTS files - historically we've avoided checking in any files that are specific to only some IDEs/workflows, such as `.vscode/` etc, so it seems like we should instead add all of these `AGENTS.md` etc. to `.gitignore` and then users can individually create symlinks to CONTRIBUTING.md if they want to use these tools.

Reply:
+1 to generalizing this with CONTRIBUTING.md